x-pack/plugins/endpoint/public/embeddables/resolver/view/defs.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo } from 'react';
+import { darken, saturate } from 'polished';
+
+import {
+  htmlIdGenerator,
+  // qualitative pallette functions.
+  //  See: https://elastic.github.io/eui/#/utilities/color-palettes
+  euiPaletteForTemperature,
+  euiPaletteForStatus,
+  colorPalette,
+} from '@elastic/eui';
+
+// Generating from `colorPalette` function: This could potentially
+//  pick up a palette shift and decouple from raw hex
+const [euiColorEmptyShade, , , , , euiColor85Shade, euiColorFullShade] = colorPalette(
+  ['#ffffff', '#000000'],
+  7
+);
+
+// A catalog of arbitraged colors
+const resolverPalette: Record<string, string | string[]> = {
+  temperatures: euiPaletteForTemperature(7),
+  statii: euiPaletteForStatus(7),
+  fullShade: euiColorFullShade,
+  emptyShade: euiColorEmptyShade,
+};
+
+/* 
+Define colors by semantics like so:
+`danger`, `attention`, `enabled`, `disabled`
+Or by function like:
+`colorBlindBackground`, `subMenuForeground` 
+*/
+type ResolverColorNames = 'ok' | 'okdark' | 'empty' | 'full' | 'warning' | 'strokeBehindEmpty';
+
+export const NamedColors: Record<ResolverColorNames, string> = {
+  ok: saturate(0.5, resolverPalette.temperatures[0]),
+  okdark: darken(0.2, resolverPalette.temperatures[0]),
+  empty: euiColorEmptyShade,
+  full: euiColorFullShade,
+  strokeBehindEmpty: euiColor85Shade,
+  warning: resolverPalette.statii[3],
+};
+
+const idGenerator = htmlIdGenerator();
+
+// To be referenced in fill and stroke attributes
+export const PaintServerIds = {
+  darkLinearReflect: idGenerator('darkreflect'),
+};
+
+// PaintServers: Where color palettes, grandients, patterns and other similar concerns
+//    are exposed to the component
+const PaintServers = memo(() => (
+  <>
+    <linearGradient
+      id={PaintServerIds.darkLinearReflect}
+      x1="-100"
+      y1="-30"
+      x2="100"
+      y2="30"
+      spreadMethod="reflect"
+      gradientUnits="userSpaceOnUse"
+    >
+      <stop offset="0%" stopColor={NamedColors.okdark} stopOpacity=".7" />
+      <stop offset="100%" stopColor={darken(0.15, NamedColors.okdark)} stopOpacity=".7" />
+    </linearGradient>
+  </>
+));
+
+// To be referenced by <use> elements
+export const SymbolIds = {
+  processNode: idGenerator('nodeSymbol'),
```

Comment:
these IDs are stable so you can just use `idGenerator('nodeSymbol')` anywhere and get the same one, right?

Reply:
Yeah IIRC if you call it with the same string, you get the same ID